megamek/src/megamek/common/AeroSpaceFighter.java

```diff
@@ -0,0 +1,179 @@
+            dealt += getSI() - finalSI;
+            setSI(finalSI);
+            setEngineHits(Math.max(3, getEngineHits()));
+        }
+
+        // Move on to actual damage...
+        int damage = getCap0Armor() - getCapArmor();
+        // Fix for #587. Only multiply if Aero Sanity is off
+        if ((null != game) && !game.getOptions().booleanOption(OptionsConstants.ADVAERORULES_AERO_SANITY)) {
+            damage *= 10;
+        }
+        damage -= dealt; // We already dealt a bunch of damage, move on.
+        if (damage < 1) {
+            return;
+        }
+        int hits = (int) Math.ceil(damage / 5.0);
+        int damPerHit = 5;
+        for (int i = 0; i < hits; i++) {
+            int loc = Compute.randomInt(4);
+            // Fix for #587. Apply in 5 point groups unless damage remainder is less.
+            setArmor(getArmor(loc) - Math.min(damPerHit, damage), loc);
+            // We did too much damage, so we need to damage the SI, but we wont
+            // reduce the SI below 1 here
+            // unless the fighter is destroyed.
+            if (getArmor(loc) < 0) {
+                if (getSI() > 1) {
+                    int si = getSI() + (getArmor(loc) / 2);
+                    si = Math.max(si, isDestroyed() || isDoomed() ? 0 : 1);
+                    setSI(si);
+                }
+                setArmor(0, loc);
+            }
+            damage -= damPerHit;
+        }
+    }
+
+    /**
+     * Damage a capital fighter's weapons. WeaponGroups are damaged by critical hits.
+     * This matches up the individual fighter's weapons and critical slots and damages those
+     * for MHQ resolution
+     * @param loc - Int corresponding to the location struck
+     */
+    public void damageCapFighterWeapons(int loc) {
+        for (Mounted weapon : weaponList) {
+            if (weapon.getLocation() == loc) {
+                //Damage the weapon
+                weapon.setHit(true);
+                //Damage the critical slot
+                for (int i = 0; i < getNumberOfCriticals(loc); i++) {
+                    CriticalSlot slot1 = getCritical(loc, i);
+                    if ((slot1 == null) ||
+                            (slot1.getType() == CriticalSlot.TYPE_SYSTEM)) {
+                        continue;
+                    }
+                    Mounted mounted = slot1.getMount();
+                    if (mounted.equals(weapon)) {
+                        hitAllCriticals(loc, i);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean isAero() {
+        return true;
+    }
+
+    /**
+     * Fighters may carry external ordnance;
+     * Other Aerospace units with cargo bays and the Internal Bomb Bay quirk may carry bombs internally.
+     * @return boolean
+     */
+    @Override
+    public boolean isBomber() {
+        return true;
+    }
+
+    @Override
+    /**
```

Comment:
I suggest not repeating any comments from superclasses; hovering the method will show the superclass javadoc

Reply:
Done.